Authentication/phpBBSessionAuthenticator.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ *
+ * @package phpBBSessionsAuthBundle
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license MIT
+ *
+ */
+namespace phpBB\SessionsAuthBundle\Authentication;
+
+
+use phpBB\SessionsAuthBundle\Tokens\phpBBToken;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\SimplePreAuthenticatorInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+
+class phpBBSessionAuthenticator implements SimplePreAuthenticatorInterface, AuthenticationFailureHandlerInterface {
+
+    /** @var  String */
+    private $cookiename;
+
+    /** @var  String */
+    private $boardurl;
+
+    /**
+     * @param $cookiename
+     * @param $boardurl
+     */
+    public function __construct($cookiename, $boardurl)
+    {
+        $this->cookiename = $cookiename;
+        $this->boardurl   = $boardurl;
+
+    }
+
+    /**
+     * @param TokenInterface $token
+     * @param UserProviderInterface $userProvider
+     * @param $providerKey
+     */
+    public function authenticateToken(TokenInterface $token, UserProviderInterface $userProvider, $providerKey)
+    {
+        // TODO: Implement authenticateToken() method.
+    }
+
+    /**
+     * @param TokenInterface $token
+     * @param $providerKey
+     * @return bool
+     */
+    public function supportsToken(TokenInterface $token, $providerKey)
+    {
+        return $token instanceof phpBBToken && $token->getProviderKey() === $providerKey;
```

Comment:
True but they are both checking for the same token to see if it is a phpBBToken. Although if this is what symfony normally does maybe let's stick to that then.